base_partner_sequence/models/partner.py

```diff
@@ -35,7 +36,7 @@ def write(self, vals):
                 and partner._needs_ref(vals=partner_vals)
                 and not partner.ref
             ):
-                partner_vals["ref"] = partner._get_next_ref(vals=partner_vals)
+                partner_vals["ref"] = partner._get_next_ref()
```

Comment:
@Nikul-OSI Why do you remove the `vals` paramter from `_get_next_ref` method? Even id it's useless in the current implementation, the `get_next_ref` method could be overridden into an other addon and build the sequence according to some infos into the vals dict. Here your change the existing api without benefits.

Reply:
Because `_get_next_ref` method, not used **vals**. https://github.com/OCA/partner-contact/pull/1453/files#diff-cec1f3044f9be2641c8e43fe9d3fb15bf0f9b957a37ee8fe908ef5b62b0fc84aR15

Reply:
@lmignon now done